### PR TITLE
WIP 230 - feat - Question Pagination

### PR DESCRIPTION
**Pull request by ramonrf**
#### What does this PR do?
- Add infinite scroll by clicking `Show More...` button.
- Add action to update question list after question edition.
- Add action to update question list after answering a question.
#### What are the relevant tickets?
WIP 230 - Add pagination to questions

### CODE REVIEW

File: src/reducers/questionsReducer.js [hunk 1]
```diff
@@ -2,12 +2,66 @@ import mapKeys from 'lodash.mapkeys';
...
+  pendingQuestions: { total: 0, items: {} },
+  resolvedQuestions: { total: 0, items: {} },
+}, action) {
   switch (action.type) {
     case QUESTIONS_GET_ALL:
-      return { ...state, items: mapKeys(action.payload.data, 'question_id') };
+      {
+        const newItems = mapKeys(action.payload.data.questions, 'question_id');
+        const prevItems = state[`${action.meta.type}Questions`].items;
+        const mergedItems = extend(prevItems, newItems);
```
Comment (alan-maldonado) [inline src/reducers/questionsReducer.js]: maybe you could use spread operator.
for example
`const mergeItems = { ...prevItems, ...newItems };`

File: src/reducers/questionsReducer.js [hunk 2]
```diff
@@ -2,12 +2,66 @@ import mapKeys from 'lodash.mapkeys';
...
+        } else if (tryToUpdateQuestion(newState, 'resolvedQuestions', action.meta.question)) {
+          return newState;
+        }
+        return state;
+      }
+    case QUESTION_ANSWER_SUBMIT:
+      {
+        const key = action.meta.question.question_id;
+        const items = state.pendingQuestions.items;
+        delete items[key];
```
Comment (alan-maldonado) [inline src/reducers/questionsReducer.js]: try not to use delete action on objects, you could use lodash.omit

File: src/actions/index.js [hunk 3]
```diff
@@ -38,12 +39,13 @@ export function dimissAlert(idx) {
   };
 }
 
-export function fetchQuestions() {
-  const request = axios.get('/api/questions', axiosConfig());
-
+export function fetchQuestions(type = 'pending', offset = 0, limit = 20) {
+  const answered = type === 'resolved';
+  const request = axios.get(`/api/questions?offset=${offset}&limit=${limit}&answered=${answered}`, axiosConfig());
```
Comment (alan-maldonado) [inline src/actions/index.js]: maybe we can make this a little clear.

```
axios.get('/user', {
    params: {
      ID: 12345
    }
  }
```
So if we want this we should pass to axiosConfig params.

File: src/reducers/questionsReducer.js [hunk 4]
```diff
@@ -1,13 +1,61 @@
...
 } from '../actions/types';
 
-export default function (state = { items: [] }, action) {
+function tryToUpdateQuestion(obj, key, question) {
+  const state = obj;
+  const subState = obj[key].items;
+  if (question.question_id in subState) {
+    subState[question.question_id].question = question.question;
+    state[key] = { ...obj[key], items: subState };
+    return true;
```
Comment (codinronan) [inline src/reducers/questionsReducer.js]: 👍

File: src/actions/index.js (hunk 3 above)
Comment (ramonrf) [inline src/actions/index.js]: I'm not sure what you mean, but must be because I'm new using `axios`but I'll look into it and make the changes.

**Pull request merged**